modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -117,75 +136,171 @@ template<typename Net> class Params {
+    */
+    Params& pluginConfig(IEConfig&& cfg) {
+        desc.config = std::move(cfg);
         return *this;
     }
 
+    /** @brief Sets count of asynchronous inference requests.
+
+    @param nireq count of inference asynchronous requests.
+    @return reference to object of class Params.
+    */
     Params& cfgNumRequests(size_t nireq) {
         GAPI_Assert(nireq > 0 && "Number of infer requests must be greater than zero!");
         desc.nireq = nireq;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
-        desc.reshape_table = std::move(reshape_table);
-        return *this;
-    }
+    /** @brief Sets new input shapes for the CNN.
 
+    The function is used to set new input shapes for network. CNN will be reshaped by
+    new shapes.
+
+    @param reshape_table map of pairs: name of corresponding data and its dimension.
+    @return reference to object of class Params.
+    */
     Params<Net>& cfgInputReshape(const std::map<std::string, std::vector<std::size_t>>& reshape_table) {
         desc.reshape_table = reshape_table;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
-        desc.reshape_table.emplace(layer_name, layer_dims);
+    /** @overload
+    Function with rvalue parameters.
+
+    @param reshape_table rvalue map of pairs: name of corresponding data and its
+    dimension.
+    @return reference to object of class Params.
+    */
+    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
+        desc.reshape_table = std::move(reshape_table);
         return *this;
     }
 
+    /** @overload
+    Function for one input layer.
+
+    @param layer_name name of layer.
+    @param layer_dims new dimensions for this layer.
+    @return reference to object of class Params.
+    */
     Params<Net>& cfgInputReshape(const std::string& layer_name, const std::vector<size_t>& layer_dims) {
         desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::unordered_set<std::string>&& layer_names) {
-        desc.layer_names_to_reshape = std::move(layer_names);
+    /** @overload
+    Function with rvalue parameters for one input layer.
+
+    @param layer_name rvalue name of layer.
+    @param layer_dims rvalue new dimensions for this layer.
+    @return reference to object of class Params.
+    */
+    Params<Net>& cfgInputReshape(std::string&& layer_name, std::vector<size_t>&& layer_dims) {
+        desc.reshape_table.emplace(layer_name, layer_dims);
         return *this;
     }
 
+    /** @overload
+    Function for reshape by image size.
+
+    The function is used to set names of layers that will be used for CNN reshape.
+    Dimensions will be constructed automatically by current CNN input and height and
+    width of image.
```

Comment:
I'd rephrase set the selected layers will be reshaped automatically its input image size

Reply:
Changed.